cmd/uid_provider/client/client.go

```diff
@@ -65,20 +66,43 @@ func NewRedisClient() *Client {
 
 // CreateUser inserts a new user into Redis
 func (c *Client) CreateUser(ctx context.Context, uid, pwd string, newUser User) error {
-	err := hasPermission(ctx, c.rdb, uid, pwd)
+	err := hasPermission(ctx, c.rdb, uid, pwd, newUser, true)
 
 	if err != nil {
 		return ierrors.NewError().Forbidden().Message(err.Error()).Build()
 	}
+
 	if err := set(ctx, c.rdb, newUser); err != nil {
 		return ierrors.NewError().BadRequest().Message(err.Error()).Build()
 	}
 	return nil
 }
 
+func isPermissionAllowed(newUserPermissionScope string, newUserPermissions []string, requestorPermissionScope string, requestorPermissions []string, isCreation bool) bool {
+	if !metautils.IsInnerScope(requestorPermissionScope, newUserPermissionScope) {
+		return false
+	}
+
+	for _, permission := range newUserPermissions {
+		if (isCreation && !utils.Includes(requestorPermissions, permission)) || !utils.Includes(requestorPermissions, auth.CreateToken) {
+			return false
+		}
+	}
+
+	return true
+}
+
```

Comment:
It'd be a good idea to move this func to the bottom of this file, so it maintains the order (first the exported functions, then the unexported ones).  
Also, as `newUserPermissionScope` and `requestorPermissionScope` are of the same type, you could just declare them like `newUserPermissionScope, requestorPermissionScope string` so it wont take that much space. The same goes to `newUserPermissions` and `requestorPermissions`

Reply:
Changed both.